src/plugins/intel_gpu/src/graph/graph_optimizer/reorder_inputs.cpp

```diff
@@ -384,28 +428,51 @@ void insert_reorders_in_dir(program& p, const std::map<program_node*, format::ty
+        //    node->get_output_layout().format : It is not up-to-date at this moment. It is just the default format (bfyx)
+        //    fmt_map.at(node).format          : It is queried with get_preferred_layout. However, it has only output format.
+        //    node.get_required_input0/output  : If it is valid(!= any), it is up-to-date. It has input format, too.
+        // So the priority is required_input0/output --> fmt_map --> output_layout().format
+
+        auto predecessor = travel_direction_wrapper<dir>::first(node, next);
+        auto successor = travel_direction_wrapper<dir>::second(node, next);
+        auto in_layout = predecessor->get_output_layout();
+        auto out_layout = in_layout;
+        in_layout.format = get_target_output_format(lo, fmt_map, predecessor);
+        auto target_input0_format = get_target_input0_format(lo, fmt_map, successor);
+
+        for (auto& fused_prim : successor->get_fused_primitives()) {
+            // If it is input of fused node, use output layout instead of input layout
+            if (successor->get_dependencies().size() <= fused_prim.dep_start_idx)
+                continue;
+            auto& dependency = successor->get_dependency(fused_prim.dep_start_idx);
+            if (&dependency == predecessor) {
+                target_input0_format = get_target_output_format(lo, fmt_map, successor);
+                GPU_DEBUG_IF(debug_config->verbose >= 2) {
+                    GPU_DEBUG_COUT << __func__ << ":" << __LINE__ << ": Use output format of successor " << successor->id() << " : "
+                                << fmt_to_str(target_input0_format) << std::endl;
+                }
+                break;
+            }
+        }
 
-        // When the input is fed into different convolutions, create separate cache entry
-        bool needs_split_reorder = false;
-        bool use_onednn_impls = lo.get_optimization_attributes().use_onednn_impls;
-        if (node->is_type<convolution>() && use_onednn_impls)
-            needs_split_reorder = lo.needs_onednn_small_ic_to_blocked(out_layout.format, in_layout, node->as<convolution>());
+        out_layout.format = target_input0_format;
+        GPU_DEBUG_IF(debug_config->verbose >= 2) {
+            GPU_DEBUG_COUT << __func__ << ":" << __LINE__ << ":" << dir_msg(dir) << "  " << node->id() << " --> " << next->id() << " ## "
+                    << fmt_to_str(in_layout.format) << " --> " << fmt_to_str(out_layout.format) << std::endl;
+        }
 
         auto reorder_pair = rf.get_reorder(travel_direction_wrapper<dir>::first(node, next)->id(),
                                            in_layout,
-                                           out_layout, needs_split_reorder);
+                                           out_layout);
         auto reorder = reorder_pair.first;
 
-        if (reorder) {
+        if (reorder && (in_layout.format != format::any && out_layout.format != format::any)) {
```

Comment:
just a question, is there any possibility that in_layout or out_layout can be format::any?

Reply:
yes, it is possible. In that case, we don't need to add reorder.